test/integration/toolchains/premake/test_premaketoolchain.py

```diff
@@ -0,0 +1,143 @@
+import os
+import textwrap
+
+from conan.test.utils.tools import TestClient
+from conan.tools.premake.toolchain import PremakeToolchain
+
+
+def test_extra_flags_via_conf():
+    profile = textwrap.dedent(
+        """
+        [settings]
+        os=Linux
+        arch=x86_64
+        compiler=gcc
+        compiler.version=9
+        compiler.cppstd=17
+        compiler.libcxx=libstdc++
+        build_type=Release
+
+        [buildenv]
+        CFLAGS=-flag00 -other=val0
+        CXXFLAGS=-flag01 -other=val1
+        LDFLAGS=-flag02 -other=val2
+
+        [conf]
+        tools.build:cxxflags=["-flag1", "-flag2"]
+        tools.build:cflags=["-flag3", "-flag4"]
+        tools.build:sharedlinkflags+=["-flag5"]
+        tools.build:exelinkflags+=["-flag6"]
+        tools.build:defines=["define1=0"]
+    """
+    )
+    tc = TestClient()
+    tc.save({"conanfile.txt": "[generators]\nPremakeToolchain", "profile": profile})
+
+    tc.run("install . -pr:a=profile")
+    content = tc.load(PremakeToolchain.filename)
+    assert 'cppdialect "c++17"' in content
+
+    assert (
+        """
+        filter {"files:**.c"}
+            buildoptions { "-flag00", "-other=val0", "-flag3", "-flag4" }
+        filter {}
+        """
+        in content
+    )
+
+    assert (
+        """
+        filter {"files:**.cpp", "**.cxx", "**.cc"}
+            buildoptions { "-flag01", "-other=val1", "-flag1", "-flag2" }
+        filter {}
+        """
+        in content
+    )
+
+    assert 'linkoptions { "-flag02", "-other=val2", "-flag5", "-flag6" }' in content
+    assert 'defines { "define1=0" }' in content
+
+
+def test_project_configuration():
+    tc = TestClient(path_with_spaces=False)
+
+    conanfile = textwrap.dedent(
+        """
+        from conan import ConanFile
+        from conan.tools.layout import basic_layout
+        from conan.tools.premake import Premake, PremakeDeps, PremakeToolchain
+        import os
+
+        class Pkg(ConanFile):
+            settings = "os", "compiler", "build_type", "arch"
+            name = "pkg"
+            version = "1.0"
+
+            def layout(self):
+                basic_layout(self, src_folder="src")
+
+            def generate(self):
+                tc = PremakeToolchain(self)
+                tc.extra_defines = ["VALUE=2"]
+                tc.extra_cflags = ["-Wextra"]
+                tc.extra_cxxflags = ["-Wall", "-Wextra"]
+                tc.extra_ldflags = ["-lm"]
+                tc.project("main").extra_defines = ["TEST=False"]
+                tc.project("test").disable = True
+                tc.project("main").extra_cxxflags = ["-FS"]
+                tc.generate()
+    """
+    )
+    tc.save({"conanfile.py": conanfile})
+    tc.run("install .")
+
+    toolchain = tc.load(
+        os.path.join("build-release", "conan", "conantoolchain.premake5.lua")
+    )
```

Comment:
```suggestion
    toolchain = tc.load("build-release/conan/conantoolchain.premake5.lua")
```
Keep it simple, this works in all platforms